src/transformers/models/big_bird/modeling_big_bird.py

```diff
@@ -1055,6 +1090,16 @@ def _get_rand_attn_plan(from_seq_length, from_block_size, num_rand_blocks):
 
         return plan_from_length, plan_num_rand_blocks
 
+    def _bigbird_block_rand_mask_fast(self, 
+            from_seq_length, to_seq_length, from_block_size, to_block_size, num_rand_blocks, last_idx, n_heads
+        ): 
+        rand_attn = [ np.zeros((from_seq_length // from_block_size - 2, num_rand_blocks), dtype=np.int32) for _ in range(n_heads)]
+        for i in range(1, from_seq_length // from_block_size - 1):
+            rand_i = np.random.randint(len(self.rand_attn_tables[i-1]), size=n_heads)
+            for j, rand_index in enumerate(rand_i):
+                rand_attn[j][i-1] = self.rand_attn_tables[i-1][rand_index]
```

Comment:
My question is that since we already have all of the permutation generated, why can't this be simpler? 
We should just need to select 1 of the permutation no? Feels like this is adding even more randomness where it is not necessarily needed.

Reply:
I tried to use `numpy.random.choice` which may be more clear in semantics. but it is much slower than `numpy.random.randint`. I guess the reason is that `numpy.random.choice` uses more instructions including generating a random integer.

Reply:
I may misunderstand what you say. The randint here is to generate a random index of the permutation table. it is what you said to select 1 of the permutation no.

Reply:
Okay, what I mean is that: 
1. We already generated all the possible permutation, so we should just select one of the permutation
2. You don't necessarily need to loop? 
```python 
rand_i = np.random.randint(len(self.rand_attn_tables[i-1]), size=n_heads)
rand_attn[j][:] = self.rand_attn_tables[:][rand_i]
```
Should work no?

Reply:
Oh, I see.  But because we pass a `size=n_heads` to `np.random.randint`, the `rand_i` will be a list rather than a single number. so, we can't do it this way.